app/models/proposal.rb

```diff
@@ -118,13 +124,15 @@ def withdraw
   end
 
   def confirm
-    transaction do
-      update!(confirmed_at: DateTime.current)
-      ps = ProgramSession.create_from_proposal(self)
-      ps.persisted?
-    end
-  rescue ActiveRecord::RecordInvalid
-    false
+    self.update(confirmed_at: DateTime.current)
+    ps = self.program_session
+    ps.update(state: self.waitlisted? ? ProgramSession::WAITLISTED : ProgramSession::LIVE) if ps.present?
```

Comment:
try inlining these two lines.  what I'd expect is something like this:

program_session.update(...)

I'd also find this line a bit more readable by leading with the if block such as:

```
if program_session.present?
  program_session.update(...)
end
```
I also found the ? inline was a bit opaque and would love to see that split off such as:

```
if program_session.present?
  new_state = waitlisted? ? ProgramSession::WAITLISTED : ProgramSession::LIVE
  program_session.update(state: new_state)
end
```